includes/class-amp-fonts.php

```diff
@@ -0,0 +1,295 @@
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Georgia',
+				'fallbacks' => [ 'Times', 'Times New Roman', 'serif' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Gill Sans',
+				'fallbacks' => [ 'Gill Sans MT', 'Calibri', 'sans-serif' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Lucida Bright',
+				'fallbacks' => [ 'Georgia', 'serif' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Lucida Sans Typewriter',
+				'fallbacks' => [ 'Lucida Console', 'monaco', 'Bitstream Vera Sans Mono', 'monospace' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Palatino',
+				'fallbacks' => [ 'Palatino Linotype', 'Palatino LT STD', 'Book Antiqua', 'Georgia', 'serif' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Papyrus',
+				'fallbacks' => [ 'fantasy' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Tahoma',
+				'fallbacks' => [ 'Verdana', 'Segoe', 'sans-serif' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Times New Roman',
+				'fallbacks' => [ 'Times New Roman', 'Times', 'Baskerville', 'Georgia', 'serif' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Trebuchet MS',
+				'fallbacks' => [ 'Lucida Grande', 'Lucida Sans Unicode', 'Lucida Sans', 'Tahoma', 'sans-serif' ],
+				'weights'   => $default_weight,
+			],
+			[
+				'name'      => 'Verdana',
+				'fallbacks' => [ 'Geneva', 'sans-serif' ],
+				'weights'   => $default_weight,
+			],
+		];
+		$file  = __DIR__ . '/data/fonts.json';
+		$fonts = array_merge( $fonts, self::get_google_fonts( $file ) );
+
+		$columns = wp_list_pluck( $fonts, 'name' );
+		array_multisort( $columns, SORT_ASC, $fonts );
+
+		$fonts_url = self::URL;
+		$subsets   = self::get_subsets();
+
+		$fonts = array_map(
+			static function ( $font ) use ( $fonts_url, $subsets ) {
+				$font['slug'] = sanitize_title( $font['name'] );
+
+				if ( ! empty( $font['gfont'] ) ) {
+					$font['handle'] = sprintf( '%s-font', $font['slug'] );
+					$font['src']    = add_query_arg(
+						[
+							'family'  => rawurlencode( $font['gfont'] ),
+							'subset'  => rawurlencode( implode( ',', $subsets ) ),
+							'display' => 'swap',
+						],
+						$fonts_url
+					);
+				}
+
+				return $font;
+			},
+			$fonts
+		);
+
+		return $fonts;
+	}
+
+	/**
+	 * Get subsets of fonts based on language settings.
+	 *
+	 * @return array
+	 */
+	public static function get_subsets() {
+		$subsets = [ 'latin', 'latin-ext' ];
+
+		/*
+		 * Translators: To add an additional character subset specific to your language,
+		 * translate this to 'greek', 'cyrillic', 'devanagari' or 'vietnamese'. Do not translate into your own language.
+		 */
+		$subset = _x( 'no-subset', 'Add new subset (greek, cyrillic, devanagari, vietnamese)', 'amp' );
+
+		if ( 'cyrillic' === $subset ) {
```

Comment:
Perhaps a `switch` here instead to make it more readable?

Reply:
This is just a copy and paste from code found elsewhere  - https://github.com/ampproject/amp-wp/pull/3998/files/f9befc5462a684639b420aa7021963d24a1db5a0#diff-4c59a5c0c9c106525335ea3d5aa2a9c6L1224-L1234

Reply:
OK, is this part still WIP and going to change? Let's keep as is if that's the case.

Reply:
This is still a PoC so it can be changed if you want.

Reply:
Changed in https://github.com/ampproject/amp-wp/pull/3998/commits/f9ed841c56fb16fa5b351d3edda1d6cf66668837